tests/drivers/build_all/modem.conf

```diff
@@ -2,9 +2,14 @@ CONFIG_TEST=y
 CONFIG_TEST_USERSPACE=y
 CONFIG_NETWORKING=y
 CONFIG_NET_IPV4=y
+CONFIG_NET_PPP=y
+CONFIG_NET_L2_PPP=y
 CONFIG_MODEM=y
 CONFIG_MODEM_HL7800=y
 CONFIG_MODEM_HL7800_FW_UPDATE=y
 CONFIG_MODEM_HL7800_SET_APN_NAME_ON_STARTUP=y
 CONFIG_MODEM_HL7800_APN_NAME="internet"
 CONFIG_MODEM_HL7800_RAT_M1=y
+CONFIG_MODEM_WNCM14A2A=y
+CONFIG_MODEM_GSM_PPP=y
+CONFIG_MODEM_GSM_UART_NAME="TEST_UART_CTLR"
```

Comment:
Looks good. Add SARA-R4 modem to test as well?

Reply:
Was lazy with SARA-R4 since it conflicts with the other modem drivers (because of networking stack).  Will add it as an explicit test.

Reply:
Is it SARA_R4 that conflicts or is it WNCM14A2A?

Reply:
Also need to add a test for HL7800

Reply:
SARA_R4 since network offload is expected from what I can tell.

Reply:
SARA_R4 uses `NET_OFFLOAD` + `NET_SOCKETS_OFFLOAD`
but WNCM14A2A also uses `NET_OFFLOAD`(no `NET_SOCKETS_OFFLOAD` in this case)